src/test/java/org/broadinstitute/hellbender/tools/picard/sam/GatherBamFilesIntegrationTest.java

```diff
@@ -67,26 +68,96 @@ public String getTestedClassName() {
+            Assert.assertNull(SamAssertionUtils.samsEqualLenient(original, outputFile, ValidationStringency.DEFAULT_STRINGENCY, referenceFile));
         } else {
-            Assert.assertNotNull(SamAssertionUtils.samsEqualLenient(original, outputFile, ValidationStringency.DEFAULT_STRINGENCY, null));
+            Assert.assertNotNull(SamAssertionUtils.samsEqualLenient(original, outputFile, ValidationStringency.DEFAULT_STRINGENCY, referenceFile));
         }
     }
+
+    @DataProvider(name="cramgathering")
+    public Object[][] gatherCRAMTestData() {
+        final File origCram = new File(TEST_DATA_DIR, "orig.cram");
+        final List<File> splitCramsUnmappedLast = Arrays.asList(
+                new File(TEST_DATA_DIR, "indchr1.cram"),
+                new File(TEST_DATA_DIR, "indchr2.cram"),
+                new File(TEST_DATA_DIR, "indchr3.cram"),
+                new File(TEST_DATA_DIR, "indchr4.cram"),
+                new File(TEST_DATA_DIR, "indchr5.cram"),
+                new File(TEST_DATA_DIR, "indchr6.cram"),
+                new File(TEST_DATA_DIR, "indchr7.cram"),
+                new File(TEST_DATA_DIR, "indchr8.cram"),
+                new File(TEST_DATA_DIR, "indUnknownChrom.cram")
+        );
+
+        final List<File> splitCramsUnmappedFirst = Arrays.asList(
+                new File(TEST_DATA_DIR, "indUnknownChrom.cram"),
+                new File(TEST_DATA_DIR, "indchr1.cram"),
+                new File(TEST_DATA_DIR, "indchr2.cram"),
+                new File(TEST_DATA_DIR, "indchr3.cram"),
+                new File(TEST_DATA_DIR, "indchr4.cram"),
+                new File(TEST_DATA_DIR, "indchr5.cram"),
+                new File(TEST_DATA_DIR, "indchr6.cram"),
+                new File(TEST_DATA_DIR, "indchr7.cram"),
+                new File(TEST_DATA_DIR, "indchr8.cram")
+        );
+
+        final List<File> splitCramsUnmappedMissing = Arrays.asList(
+                new File(TEST_DATA_DIR, "indchr1.cram"),
+                new File(TEST_DATA_DIR, "indchr2.cram"),
+                new File(TEST_DATA_DIR, "indchr3.cram"),
+                new File(TEST_DATA_DIR, "indchr4.cram"),
+                new File(TEST_DATA_DIR, "indchr5.cram"),
+                new File(TEST_DATA_DIR, "indchr6.cram"),
+                new File(TEST_DATA_DIR, "indchr7.cram"),
+                new File(TEST_DATA_DIR, "indchr8.cram")
+        );
+
+        return new Object[][]{
+                // these require the fix to https://github.com/samtools/htsjdk/issues/365
+                // in PR https://github.com/samtools/htsjdk/pull/368
+                //{splitCramsUnmappedLast, origCram, true, true},
+                //{splitCramsUnmappedFirst, origCram, true, true},
+                //{splitCramsUnmappedMissing, origCram, false, false}
+        };
+    }
+
+    // These tests require the fix to https://github.com/samtools/htsjdk/issues/365
```

Comment:
I would prefer to have those disabled by `enabled=false`. Please enter a ticket for gatk as well - to remember to reenable them